plugins/obs-outputs/CMakeLists.txt

```diff
@@ -1,200 +1,182 @@
-		${obs-outputs_PLATFORM_DEPS}
-		w32-pthreads)
-endif()
+  target_sources(obs-outputs PRIVATE obs-outputs.rc)
 
-if(APPLE AND (WITH_RTMPS OR (WITH_RTMPS STREQUAL "AUTO")))
-	find_library(FOUNDATION_FRAMEWORK Foundation)
-	find_library(SECURITY_FRAMEWORK Security)
+  if(MSVC)
+    target_link_libraries(obs-outputs PRIVATE OBS::w32-pthreads)
+  endif()
 
-	set(obs-outputs_PLATFORM_DEPS
-		${obs-outputs_PLATFORM_DEPS}
-		${FOUNDATION_FRAMEWORK}
-		${SECURITY_FRAMEWORK})
+  target_link_libraries(obs-outputs PRIVATE ws2_32 winmm Iphlpapi)
 endif()
 
-set(obs-outputs_librtmp_HEADERS
-	librtmp/amf.h
-	librtmp/bytes.h
-	librtmp/cencode.h
-	librtmp/dh.h
-	librtmp/dhgroups.h
-	librtmp/handshake.h
-	librtmp/http.h
-	librtmp/log.h
-	librtmp/md5.h
-	librtmp/rtmp.h
-	librtmp/rtmp_sys.h)
-set(obs-outputs_librtmp_SOURCES
-	librtmp/amf.c
-	librtmp/cencode.c
-	librtmp/hashswf.c
-	librtmp/log.c
-	librtmp/md5.c
-	librtmp/parseurl.c
-	librtmp/rtmp.c)
-
-if(NOT WIN32)
-	set_source_files_properties(${obs-outputs_librtmp_SOURCES} PROPERTIES
-		COMPILE_FLAGS "-fvisibility=hidden")
+if(ENABLE_RTMPS STREQUAL "AUTO" OR ENABLE_RTMPS STREQUAL "ON")
+  find_package(MbedTLS)
+  find_package(ZLIB)
+  if(NOT MBEDTLS_FOUND OR NOT ZLIB_FOUND)
+    if(ENABLE_RTMPS STREQUAL "ON")
+      message(
+        FATAL_ERROR
+          "OBS: mbedTLS or zlib not found, but required for RTMPS support.")
+      return()
+    else()
+      message(
+        WARNING
+          "OBS: mbedTLS or zlib was not found, RTMPS will be auto-disabled")
+      target_compile_definitions(obs-outputs PRIVATE NO_CRYPTO)
+    endif()
+  else()
+    target_sources(
+      obs-outputs
+      PRIVATE librtmp/amf.c
+              librtmp/amf.h
+              librtmp/bytes.h
+              librtmp/cencode.c
+              librtmp/cencode.h
+              librtmp/dh.h
+              librtmp/dhgroups.h
+              librtmp/handshake.h
+              librtmp/hashswf.c
+              librtmp/http.h
+              librtmp/log.c
+              librtmp/log.h
+              librtmp/md5.c
+              librtmp/md5.h
+              librtmp/parseurl.c
+              librtmp/rtmp.c
+              librtmp/rtmp.h
+              librtmp/rtmp_sys.h)
+
+    target_compile_definitions(obs-outputs PRIVATE USE_MBEDTLS CRYPTO)
+
+    target_link_libraries(obs-outputs PRIVATE Mbedtls::Mbedtls ZLIB::ZLIB)
+
+    if(OS_WINDOWS)
+      target_link_libraries(obs-outputs PRIVATE crypt32)
+
+    elseif(OS_MACOS)
+      find_library(FOUNDATION_FRAMEWORK Foundation)
+      find_library(SECURITY_FRAMEWORK Security)
+      mark_as_advanced(FOUNDATION_FRAMEWORK SECURITY_FRAMEWORK)
+
+      target_link_libraries(obs-outputs PRIVATE ${FOUNDATION_FRAMEWORK}
+                                                ${SECURITY_FRAMEWORK})
+      set_target_properties(obs-outputs PROPERTIES CXX_VISIBILITY_PRESET hidden)
+
+    elseif(OS_POSIX)
+      set_target_properties(obs-outputs PROPERTIES CXX_VISIBILITY_PRESET hidden)
+    endif()
+  endif()
 endif()
```

Comment:
Between those two `endif()`, I think this might be needed if we we set `ENABLE_RTMPS` to `OFF`:
```cmake
else()
  target_compile_definitions(obs-outputs PRIVATE NO_CRYPTO)
```

Reply:
Yeah still have to fix that.